pallets/aleph/src/lib.rs

```diff
@@ -121,6 +161,49 @@ pub mod pallet {
         pub(crate) fn set_next_emergency_finalizer(emergency_finalizer: T::AuthorityId) {
             <NextEmergencyFinalizer<T>>::put(emergency_finalizer);
         }
+
+        pub(crate) fn current_session() -> u32 {
+            T::SessionInfoProvider::current_session()
+        }
+
+        // If a scheduled future version change is rescheduled to a different session,
+        // it is possible to reschedule it with the same version as initially.
+        // To cancel a future version change, reschedule it with the current version.
+        // If a scheduled version change has moved into the past, `SessionManager` records it
+        // as the current version.
+        pub(crate) fn schedule_next_aleph_bft_version_change(
+            version_change: VersionChange,
+        ) -> Result<(), &'static str> {
```

Comment:
Is there a specific reason we use strings here and not an enum?

Reply:
This is due to `DispatchResult` being returned later on. If we do not want to include this information in the error variant, then it would be possible to get rid of the string, but the error itself would be less informative.

Reply:
Ok. I see now. This is a helper functional that is called internally only. 

I think it has a slightly confusing name actually :) There is naming convention in substrate pallets (and much broader) for such functions: if the call is `something` then the internal function is called `do_something`. In this case this should be called `do_schedule_aleph_bft_version_change`. 

Feel free to incorporate if it makes sense and resolve all my comments.

Reply:
Sure, `do` sounds much better then `next` in this context.